src/software/thunderscope/replay/test/replay_corruption_test.py

```diff
@@ -0,0 +1,189 @@
+"""
+TO GIVE CONTEXT FOR THE THIS TEST: 
+
+The idea of this test is to check how robust proto player is against corrupted file entries. 
+We are going to test how robust proto player is by intentionally creating corrupted replay files 
+and seeing whether or not proto player could play the replay files. 
+
+Steps to test: 
+    1. we create a directory to store the replay files 
+    2. we create invalid entries of two type: the data is actually corrupted, and we are missing a delimeter. We 
+        the mixed those replay entries with valid replay entries to check if proto player could actually player those
+        entires
+    4. we check to see if there are uncaught exception. If there are, we know fore sure proto player wouldn't work!
+"""
+import pytest
+import sys 
+
+import time
+from typing import Callable
+import random
+import shutil
+import gzip
+import os
+from proto.import_all_protos import *
+from google.protobuf.message import Message
+import base64
+from software.thunderscope.replay.replay_constants import *
+from software.thunderscope.replay.proto_player import ProtoPlayer
+from software.thunderscope.replay.proto_logger import ProtoLogger
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+
+# this would be the path where the file corruption would be tested
+TMP_REPLAY_SAVE_PATH = "/tmp/test_file_corruption"
+
+def create_random_proto() -> Message:
+    """
+    creating referencing protos 
```

Comment:
not sure what you mean by this description, I'll suggest to update this

Reply:
done.